src/eth_provider/provider.rs

```diff
@@ -131,6 +133,8 @@ pub trait EthereumProvider {
     ) -> EthProviderResult<FeeHistory>;
     /// Send a raw transaction to the network and returns the transactions hash.
     async fn send_raw_transaction(&self, transaction: Bytes) -> EthProviderResult<B256>;
+    /// Check the cached balance of the signer and return an error if the balance is not enough.
+    async fn check_cached_balance(&self, signer: Address, transaction_value: U256) -> Result<(), EthApiError>;
```

Comment:
can we move this out of the trait? The trait is more to define the behaviour of a EthProvider, it shouldn't be in there